### DISCUSSION TITLE
How to get OpenGL 4.3+ support using nvidia drivers?

### DISCUSSION THREAD

**Discussion by jessey-git** [microsoft/wslg#1086] [General]
Recently a project I contribute to has moved to requiring a minimum of OpenGL 4.3.  Unfortunately I only seem able to get up to 4.2 using nvidia drivers, even though the hardware is capable of full 4.6:

```
Extended renderer info (GLX_MESA_query_renderer):
    Vendor: Microsoft Corporation (0xffffffff)
    Device: D3D12 (NVIDIA GeForce GTX 1070 with Max-Q Design) (0xffffffff)
    Version: 22.2.5
    Accelerated: yes
    Video memory: 24394MB
    Unified memory: no
    Preferred profile: core (0x1)
    Max core profile version: 4.2
    Max compat profile version: 4.2
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL vendor string: Microsoft Corporation
OpenGL renderer string: D3D12 (NVIDIA GeForce GTX 1070 with Max-Q Design)
OpenGL core profile version string: 4.2 (Core Profile) Mesa 22.2.5-0ubuntu0.1~22.04.3
OpenGL core profile shading language version string: 4.20
```

Do I have any options forward here besides using `LIBGL_ALWAYS_SOFTWARE=1` which "works" but is so slow it might as well not.

**Comment by onomatopellan** [comment]
Gpu acceleration in WSL2 is done by the d3d12 driver from mesa and is limited to OpenGL 4.2 for now. You can follow the progress here:

https://mesamatrix.net/#OpenGL
